frontend/src/concepts/pipelines/content/pipelinesDetails/pipelineRun/useFetchRunArtifact.tsx

```diff
@@ -7,10 +7,18 @@ import {
 import useFetchState, { FetchState } from '~/utilities/useFetchState';
 import { usePipelinesAPI } from '~/concepts/pipelines/context';
 
-export const useFetchRunArtifact = (runId: string): FetchState<Artifact[]> => {
+export const useFetchRunArtifact = (
+  runId: string,
+  modelRegistryAvailable: boolean,
```

Comment:
dont tie this to model registry, allow runId to be optional and throw a not ready error when undefined. then conditionally pass the run id into the hook based on modelRegistryAvailable